parsers/verilog.c

```diff
@@ -873,12 +873,30 @@ static int _readWordToken (tokenInfo *const token, int c, bool skip)
 	Assert (isWordToken (c));
 
 	clearToken (token);
-	do
+	if (c == '\\')
 	{
-		vStringPut (token->name, c);
-		c = vGetc ();
-	} while (isIdentifierCharacter (c));
-	_updateKind (token);
+		// Escaped identifier
+		c = vGetc (); // skip leading '\'
+		// A single `\` would result in an empty identifier, which is unsupported.
+		// Add the initial `\` in that case, and also in case it starts with `\\`.
+		if (!isgraph (c) || c == '\\')
+			vStringPut (token->name, '\\');
+		while (isgraph (c))
```

Comment:
When I tried to Merge this PR, I remembered this.

@cousteaulecommandant,

How do you think?

Reply:
> Considering locales excepting C, I think `isgraph()` can not be used here.

`isalpha()`, `isalnum()`, and `isspace()` are used throughout the Verilog parser, and are also affected by locales.  I don't see why `isgraph()` would be any different, and would have assumed it is preferred for consistency with the rest of the parser, which uses the `is*()` functions rather than checking the values explicitly.

> The isalpha function tests for any character for which isupper or islower is true, or any character that is one of a **locale-specific** set of alphabetic characters for which none of iscntrl, isdigit, ispunct, or isspace is true.

> The isspace function tests for any character that is a standard white-space character or is one of a **locale-specific** set of characters for which isalnum is false.

If you think this family of functions shouldn't be used at all just in case non-ASCII characters or messy locales are present, then I think it should be fixed *for the whole parser*, in a separate PR.  The purpose of this PR was to introduce a new feature; I don't think it's within its scope to fix a different problem that was already there.

So, for consistency with the rest of the coding style in the project, I think it's best to use `isgraph()` here, since it's the approach the rest of the parser seems to use; and perhaps fix it along with the rest in a future commit.  But I can make my piece of code an exception and make it use explicit ASCII numeric boundaries if you prefer (which, in my opinion, is less readable and maintainable).

Reply:
I totally agree with you on each of your comments.

I got curious about locale after reading the C specification you cited. Sorry for bothering you.


I've also checked the LRM about locale.  What I could only found was;

> A.9.2 Comments
> ...
> comment_text ::= { Any_ASCII_character }

The LRM does not consider locale at all even in comments.
If we would consider locale, we should rather use `isgraph()` here.

Reply:
OR, we set the locale to C for parsing.  Is that feasible?  Parsing should be locale-independent anyway, and not be affected by the user's locale.
(Or more specifically, some languages such as Python do support Unicode characters, and others such as Verilog do not, so we could control that.)